### PR TITLE
dump_only=True -> readOnly

### LINKED ISSUE #39: Setting dump_only=True in a Schema field should result in readOnly in the spec
See https://github.com/marshmallow-code/apispec/issues/84 for an example of a competing library adding this feature.

### PR DESCRIPTION
**Pull request by twosigmajab**
Fixes #39.

### CODE REVIEW

**Comment by twosigmajab** [comment]
Here is a screenshot of the Swagger UI resulting from the app below with these changes:

screen shot 2018-11-27 at 16 57 32

Note how the default value of the textarea doesn't get the dump_only "id" field added to the example Todo, as it did before these changes. You can also see "readOnly: true" when you expand the Todo model.

```python
from flask_rebar import Rebar
from flask import Flask
from marshmallow import Schema, fields
from marshmallow.validate import Length

app = Flask(__name__)
rebar = Rebar()
registry = rebar.create_handler_registry()


class Todo(Schema):
    id = fields.Int(dump_only=True)
    task = fields.Str(validate=Length(min=1, max=255))


@registry.handles(
    rule='/todos/',
    method='POST',
    request_body_schema=Todo(),
)
def create_todo():
    return None, 201


rebar.init_app(app)
```

File: tests/swagger_generation/test_marshmallow_to_swagger.py [hunk 1]
```diff
@@ -93,6 +93,27 @@ class Foo(m.Schema):
             }
         )
 
+    def test_dump_only_read_only(self):
```
Comment (barakalon) [inline tests/swagger_generation/test_marshmallow_to_swagger.py]: I think you can save some lines and move this to `test_primitive_types`:

Add an item like this to the list:
```
(m.fields.Integer(dump_only=True), {'type': 'integer', 'readOnly': True}),
```

Comment (twosigmajab) [reply tests/swagger_generation/test_marshmallow_to_swagger.py]: Good call, thanks. Done in the latest revision.

**Comment by twosigmajab** [comment]
Thanks for approving @barakalon! Does this look okay to you, @juliusiv?

**Comment by twosigmajab** [comment]
Thanks for merging all these @brockhaywood! Would it be an ok time to cut a new release now that these have landed? 😊

**Comment by twosigmajab** [comment]
@brockhaywood Noticed you already have [travis set up](https://github.com/plangrid/flask-rebar/blob/master/.travis.yml#L33) to deploy to PyPI for you, nice! I think it should just be a matter of `git tag -a "v1.2.0" -m "tag v1.2.0" && git push --tags`, and then sitting back and letting Travis handle the rest?

**Comment by barakalon** [comment]
https://github.com/plangrid/flask-rebar/blob/master/CONTRIBUTING.rst#releasing-to-pypi

**Comment by twosigmajab** [comment]
Oh nice. Just noticed https://travis-ci.org/plangrid/flask-rebar/builds/461477143 ran about an hour ago for a new v1.2.0 *branch* and failed. Any idea what went wrong?

**Comment by brockhaywood** [comment]
@twosigmajab yea, i just screwed up and accidentally bumped twice. made a new branch am rebuilding.

**Comment by brockhaywood** [comment]
@twosigmajab 1.2.0 is published!

**Comment by twosigmajab** [comment]
Awesome, thanks @brockhaywood! 🎁 ✨ 😻

**Pull request merged**